scim/usersRecords.bal

```diff
@@ -0,0 +1,356 @@
+# + userName - unique identifier for the user 
+# + name - name of the user  
+# + displayName - name displayed to end-users
+# + nickName - casual name of the user  
+# + profileUrl - URI of the user's online profile  
+# + emails - email addresses for the User  
+# + addresses - physical mailing address for the user  
+# + phoneNumbers - phone numbers for the user  
+# + photos - URI of image of the user  
+# + locale - default location of the user  
+# + active - administrative status of the user 
+# + meta - metadata of the user 
+# + urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User - attributes belong to business or enterprise extension schema  
+# + urn\:scim\:wso2\:schema - attributes belong to WSO2 extension schema  
+# + roles - list of roles for the user
+public type UserResource record {
+    string[]|string schemas?;
+    string id?;
+    int|string externalId?;
+    string userName?;
+    Name name?;
+    string displayName?;
+    string nickName?;
+    string profileUrl?;
+    string[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Photo[] photos?;
+    string locale?;
+    boolean active?;
+    Meta meta;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    WSO2 urn\:scim\:wso2\:schema?;
+    Role[] roles?;
+};
+
+# + display - human readable name, primarily used for display purposes
+# + value - label representing a collection of entitlements  
+# + \$ref - reference URI 
+# + resourceType - resource type of the entitlement
+public type Role record {
+    string display;
+    string value?;
+    string \$ref?;
+    string resourceType?;
+};
+
+# + created - DateTime that the resource was added 
+# + location - URI of the resource being returned 
+# + lastModified - most recent DateTime that the details of this resource were updated  
+# + resourceType - name of the resource type of the resource  
+# + 'version - version of the resource being returned
+public type Meta record {
+    string created;
+    string location;
+    string lastModified;
+    string resourceType?;
+    string 'version?;
+};
+
+# + formatted - full name, including all middle names, titles, and suffixes as appropriate, formatted for display  
+# + givenName - given name of the User  
+# + familyName - family name of the User  
+# + middleName - middle name(s) of the User 
+# + honorificPrefix - title 
+# + honorificSuffix - suffix
+public type Name record {
+    string formatted?;
+    string givenName?;
+    string familyName?;
+    string middleName?;
+    string honorificPrefix?;
+    string honorificSuffix?;
+};
+
+# + idpType - identity provider type  
+# + isReadOnlyUser - whether the user is read only
+# + userSource - source of the user 
+# + photoUrl - URI of the user's photo  
+# + userAccountType - type of the user account  
+# + askPassword - whether the user is asked to provide a password
+public type WSO2 record {
```

Comment:
why do we have this as `WSO2` ?

Reply:
It was used to define the attributes in this schema "urn\:scim\:wso2\:schema?"